### PR TITLE
Localization support

### PR DESCRIPTION
**Pull request by SneakyOttersec**
Adding support for regional label (thi is up for the user to choose a good one)
Adding randomized unicode badchar selection

In case you found that irrelevant / buggy please tell me, this is my first pull request ever !

Cheers

N.B : Better with the proper version

### CODE REVIEW

File: WorkbookStream.cs [hunk 1]
```diff
@@ -321,26 +321,36 @@ public List<BOF> GetMacroSheetBOFs()
...
+            }
+            replaceLabelStringLbl.SetName(new XLUnicodeStringNoCch(unicodeLabelWithBadChars, true));
             replaceLabelStringLbl.fBuiltin = false;
 
             // Hidden removes from the label manager entirely, but doesn't seem to work if fBuiltin is false
             // replaceLabelStringLbl.fHidden = true;
 
             WorkbookStream obfuscatedStream = ReplaceRecord(autoOpenLbl, replaceLabelStringLbl);
             obfuscatedStream = obfuscatedStream.FixBoundSheetOffsets();
+            Console.WriteLine(obfuscatedStream);
```
Comment (michaelweber) [inline WorkbookStream.cs]: I'd recommend commenting this out / taking this debug writeline out, but besides that this looks good!

File: WorkbookStream.cs [hunk 2]
```diff
@@ -321,26 +321,36 @@ public List<BOF> GetMacroSheetBOFs()
...
             List<Lbl> labels = GetAllRecordsByType<Lbl>();
             Lbl autoOpenLbl = labels.First(l => l.fBuiltin && l.Name.Value.Equals("\u0001") ||
-                                                l.Name.Value.ToLower().StartsWith("auto_open"));
+                                                l.Name.Value.ToLower().StartsWith(localizedLabel));
             Lbl replaceLabelStringLbl = ((BiffRecord)autoOpenLbl.Clone()).AsRecordType<Lbl>();
 
             //Characters that work
             //fefe, ffff, feff, fffe, ffef, fff0, fff1, fff6, fefd, 0000, dddd
             //Pretty much any character that is invalid unicode - though \ucccc doesn't seem to work - need better criteria for parsing
             //TODO [Stealth] Randomize which invalid unicode characters are injected into this string
```
Comment (michaelweber) [inline WorkbookStream.cs]: You went and implemented this, so we can remove it =)

**Review by michaelweber — CHANGES_REQUESTED** [review]
Thanks for helping to improve the project!

Only change I'd request before we merge it is that you drop the `Console.WriteLine` in the auto open label generation portion. Also you implemented one of the TODOs and we can remove that comment =).

This is an excellent change though - glad to see someone else was able to read the code well enough to know where to do this.

File: WorkbookStream.cs (hunk 1 above)
Comment (SneakyOttersec) [reply WorkbookStream.cs]: Should be removed now !

File: WorkbookStream.cs (hunk 2 above)
Comment (SneakyOttersec) [reply WorkbookStream.cs]: Comment removed

**Comment by SneakyOttersec** [comment]
Thanks for the quick reply on this PR and  glad to help this project ! we should be good to go :)

**Review by michaelweber — APPROVED** [review]
Fantastic, thanks for your contribution!

File: WorkbookStream.cs [hunk 3]
```diff
@@ -321,19 +321,27 @@ public List<BOF> GetMacroSheetBOFs()
...
+        public WorkbookStream ObfuscateAutoOpen(string localizedLabel)
         {
+            Random randomUnicodeChar = new Random();
+            string[] badUnicodeChars = { "\ufefe", "\uffff", "\ufeff", "\ufffe", "\uffef", "\ufff0", "\ufff1", "\ufff6", "\ufefd", "\u0000", "\udddd" };
+            int indexLabel = 0;
+            string unicodeLabelWithBadChars = "";
             List<Lbl> labels = GetAllRecordsByType<Lbl>();
             Lbl autoOpenLbl = labels.First(l => l.fBuiltin && l.Name.Value.Equals("\u0001") ||
-                                                l.Name.Value.ToLower().StartsWith("auto_open"));
+                                                l.Name.Value.ToLower().StartsWith(localizedLabel));
```
Comment (decalage2) [inline WorkbookStream.cs:332]: You might need to convert localizedLabel to lowercase here.

**Pull request merged**